src/zcl_abapgit_tadir.clas.abap

```diff
@@ -98,10 +100,18 @@ CLASS zcl_abapgit_tadir IMPLEMENTATION.
     ENDIF.
 
     IF NOT io_dot IS INITIAL.
-      lv_path = zcl_abapgit_folder_logic=>package_to_path(
+      "Reuse given Folder Logic Instance
+      lo_folder_logic = io_folder_logic.
+      IF lo_folder_logic IS NOT BOUND.
+        "Get Folder Logic Instance
+        lo_folder_logic = zcl_abapgit_folder_logic=>get_instance( ).
+      ENDIF.
+
+      lv_path = lo_folder_logic->package_to_path(
         iv_top     = iv_top
         io_dot     = io_dot
-        iv_package = iv_package ).
+        iv_package = iv_package
+        io_folder_logic = lo_folder_logic ). "Hand down existing folder logic instance
```

Comment:
this casues syntax error

lo_folder_logic should be passed to the method build in the recursion, instead of to lo_folder_logic intself

Reply:
You are correct. Must have gotten mixed up when I separated the PRs and copied the code... 😞